### PR TITLE
Refine the lintstore section

### PR DESCRIPTION
**Pull request by tshepang**
cc #1414

### CODE REVIEW

**Comment by tshepang** [comment]
@JohnTitor I do not understand the date-check sentence, so please bump it if it's still current (and maybe explain it some more):

> As of February 2022, we lint against direct declarations without the use of the macro today (although this may change in the future, as the macro is somewhat unwieldy to add new fields to, like all macros).

**Comment by JohnTitor** [comment]
> @JohnTitor I do not understand the date-check sentence, so please bump it if it's still current (and maybe explain it some more).
> 
> Here it is:
> 
> > As of February 2022, we lint against direct declarations without the use of the macro today (although this may change in the future, as the macro is somewhat unwieldy to add new fields to, like all macros).

I think the "we lint~" part and the "although this~" part talk about different things. The first says linting by rustc doesn't work on macros. The second says we use the declare_lint macro, but it may be changed in the future. The original paragraph is here: https://github.com/rust-lang/rustc-dev-guide/pull/476/files#diff-8f8ad089b3a6deea75fb8bc35716ff0b43bb6bfaf16e639afac64ed033a8ef37R16-R20

The date reference is for the first, so I think the situation still continues and we could just update the date but we could also improve the wording, I also think it's kinda confusing.

File: src/diagnostics/lintstore.md [hunk 1]
```diff
@@ -8,14 +9,15 @@ everything rotates. It's not available during the early parts of compilation
...
+
 There are two parts to the linting mechanism within the compiler: lints and
 lint passes. Unfortunately, a lot of the documentation we have refers to both
 of these as just "lints."
 
 First, we have the lint declarations themselves: this is where the name and
 default lint level and other metadata come from. These are normally defined by
 way of the [`declare_lint!`] macro, which boils down to a static with type
-`&rustc_session::lint::Lint`.
+`&rustc_lint::Lint`.
```
[the remark this answers is not in the document]
Comment (tshepang) [reply src/diagnostics/lintstore.md:20]: Oh kool, didn't think of link checker.
Also, did not notice that `rust_lint::Lint` is a re-export... thanks.

**Comment by tshepang** [comment]
that seems to conflict with what is written there, but I pushed the update with the new date regardless

**Comment by JohnTitor** [comment]
I mean, we should do rewording something like:

```
First, we have the lint declarations themselves: this is where the name and default lint level and other metadata come from. These are normally defined by way of the [declare_lint!](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_session/macro.declare_lint.html) macro, which boils down to a static with type [&rustc_lint_defs::Lint](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint_defs/struct.Lint.html) (although this may change in the future, as the macro is somewhat unwieldy to add new fields to, like all macros).

As of Aug 2022, we lint against direct declarations without the use of the macro.
```

**Comment by tshepang** [comment]
@JohnTitor I pushed

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks!

**Comment by anirudh24seven** [comment]
I think 7b5c503 did not remove the previous paragraph while adding the new paragraph. Raising a PR to remove the near-duplicate paragraph.

**Pull request merged**